tls/s2n_resume.c

```diff
@@ -349,16 +376,10 @@ S2N_RESULT s2n_deserialize_resumption_state(struct s2n_connection *conn, struct
         if (conn->mode == S2N_SERVER) {
             RESULT_GUARD_POSIX(s2n_tls12_deserialize_resumption_state(conn, from));
         } else {
-            RESULT_GUARD(s2n_tls12_client_deserialize_session_state(conn, from));
+            RESULT_GUARD(s2n_tls12_client_deserialize_session_state(conn, ticket, from));
         }
     } else if (format == S2N_SERIALIZED_FORMAT_TLS13_V1) {
-        RESULT_GUARD(s2n_tls13_deserialize_session_state(conn, psk_identity, from));
-        if (conn->mode == S2N_CLIENT) {
-            /* Free the client_ticket after setting a psk on the connection.
-             * This prevents s2n_connection_get_session from returning a TLS1.3
-             * ticket before a ticket has been received from the server. */
-            RESULT_GUARD_POSIX(s2n_free(&conn->client_ticket));
```

Comment:
Why are you getting rid of the mechanism that frees the memory for the client ticket in TLS1.3? Doesn't this mean that in TLS1.3 you could get a ticket before receiving one from the server?

Reply:
I'm getting rid of the mechanism to free the client ticket in TLS1.3, but I'm also getting rid of the mechanism to set the client ticket in TLS1.3 ;) Previously this logic was required because s2n_client_deserialize_with_session_ticket set conn->client_ticket, but that no longer happens. s2n_client_deserialize_with_session_ticket just sets up a new blob representing the ticket, and passes it on for the individual TLS1.2 / TLS1.3 deserialize methods to deal with.